File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    answer = []
    n = len(sequence)
    start, end = 0, 0
    curr_sum = 0
    min_length = float('inf')

    while end < n:
        curr_sum += sequence[end]
        
        while start <= end and curr_sum > k:
            curr_sum -= sequence[start]
            start += 1
       
        if curr_sum == k:
            curr_length = end - start + 1
            if curr_length < min_length:
                min_length = curr_length
                answer = [start,end]
          
        end += 1
    
    
        
    return answer

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = new int[2];
        
        int start = 0;
        int sum = 0;
        int minLen = Integer.MAX_VALUE;
        int endIdx = 0;
        int minStart = Integer.MAX_VALUE;
        
        
        for (int end = 0; end < sequence.length; end++) {
            sum += sequence[end];
            
            while (sum >= k) {
                if (sum == k) { 
                  int currLen = end - start + 1;
                  if (currLen < minLen) {
                    minLen = currLen;
                    answer[0] = start;
                    answer[1] = end;
                    }
                }
                sum -= sequence[start];
                start++;
            }
        }
        

        return answer;
    }
}